### ISSUE TITLE
Fix Acceptance tests that use SimulatedException [v6]

### ISSUE THREAD

**Issue by tmasternak** [Particular/NServiceBus#3059]
Currently our acceptance tests (not sure if all or only some) assume that SimulatedExeption thrown after send or sendlocal result in following behavior:
1. Message reaches transport sender
2. Send operation in performed conforming to whatever transnational settings were specified.
3. Exception is thrown
4. Receiving transaction is rolled-back
Due to message batching and outbox this is no longer true. What happens is this:
1. Message never reaches transport but is stored in pending operations collection instead
2. Exception is thrown
3. Receiving transaction is rolled-back and collection cleared

In other words some of tests do now test what we think they test (hence the impact). The simplest solution is to use pipeline behavior which runs after TransportReceiveToPhysicalMessageProcessingConnector and throws.
The downside of such solution is that tests will be harder to understand than they are now.

/cc: @andreasohlund @SzymonPobiega @danielmarbach

**Comment by tmasternak** [comment]
As we need those changes fro (sql transport v3)[https://github.com/Particular/NServiceBus.SqlServer/pull/94] it looks that we should do it as soon as possible and start with "blow-up" behavior.

/cc: @andreasohlund @weralabaj

**Comment by weralabaj** [comment]
@tmasternak can you add the sample test name where that was an issue?

**Comment by tmasternak** [comment]
This is one of [them](https://github.com/Particular/NServiceBus/blob/058d42cfc66ef51a75c50d8440d0e9d3235e0fbe/src/NServiceBus.AcceptanceTests/Tx/When_receiving_with_native_multi_queue_transaction.cs) but there are potentially ~10 more (all places where send and throw simulatedexception are in the same handler).

**Comment by timbussmann** [comment]
I'm having difficulties to understand the issue.

> 4. Receiving transaction is rolled-back

and

> 3. Receiving transaction is rolled-back and collection cleared

that sounds fine to me? Why should that test care whether the message has actually been rolled back or if it has been purged from the collection?

why is that related to SimulatedException? It could be any Exception which is thrown in message handlers.

**Comment by timbussmann** [comment]
regarding the testing of the transaction behavior, is that something which is covered by the new transport tests @andreasohlund ?

**Comment by andreasohlund** [comment]
Indeed, the seam tests should cover this, please re-open if you feel otherwise @tmasternak
